### DISCUSSION TITLE
Please clarify usage limits for public servers

### DISCUSSION THREAD

**Discussion by Lilianne-Blaze** [rustdesk/rustdesk#10254] [Q&A]
1. Do I understand it correctly that there is a limit of 300 clients connecting from a single ip per day? That there are no long-term limits like X per ip total, or X per ip per year?
2. Do I understand it correctly that there is a limit of 30 "tell me which ids are visible" requests per 60 sec, and that exceeding it blocks the client for 60 sec, with no other long term limits / blocks?

3. I need to have some way to see what ids become visible / invisible from other programs, I already made some changes to write it to logs so it can be read and parsed, but I noticed a normal client makes only 3 requests at 20 sec intervals when the window becomes active, is that correct? What if I just changed that limit to de-facto unlimited requests, so they're sent until the window is closed, I see no reason why it wouldn't work, but would it be ok / allowed use? Assuming only 1-3 clients with that mod would be running from one ip, and the limits from 1 and 2 won't be exceeded? If not, would it be ok if the interval was increased to 1-5 min?

**Comment by rustdesk** [comment]
https://www.reddit.com/r/rustdesk/comments/1cr8kfv/should_you_selfhost_a_rustdesk_server/

**Reply by Lilianne-Blaze** [reply]
That's not an answer for my questions, and in plenty of scenarios selfhosting simply isn't possible for organizational reasons.
